tfjs-backend-webgl/src/kernels/BitwiseAnd.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, BitwiseAnd, BitwiseAndInputs, DataTypeMap, env, KernelConfig, KernelFunc, TensorInfo, TypedArray, upcastType} from '@tensorflow/tfjs-core';
+import {MathBackendWebGL} from '../backend_webgl';
+import {BinaryOpProgram} from '../binaryop_gpu';
+import {BinaryOpPackedProgram} from '../binaryop_packed_gpu';
+import {bitwiseAndImplCPU as cpuBitWiseAnd} from '../kernel_utils/shared';
+
+export const BITWISEAND = `
+  int r = int(a.r) & int(b.r);
+  int g = int(a.g) & int(b.g);
+  int rb = int(a.b) & int(b.b);
+  int ra = int(a.a) & int(b.a);
+  return vec4(r, g, rb, ra);
+`;
+
+export const BITWISEAND_UNPACKED = `
+  return float(int(a.r) & int(b.r));
+`;
+
+export function bitwiseAnd(args: {
+  inputs: BitwiseAndInputs,
+  backend: MathBackendWebGL,
+  dtype?: keyof DataTypeMap
+}): TensorInfo {
+  const {inputs, backend, dtype} = args;
+  const {a, b} = inputs;
+  const webglBackend = backend as MathBackendWebGL;
+  const shouldUsePackedProgram =
+      env().getBool('WEBGL_PACK_BINARY_OPERATIONS') && BITWISEAND != null;
```

Comment:
Nit: ` && BITWISEAND != null`: This should never be null, right, since it's defined above?